### ISSUE TITLE
CondaEnvironment task should support conda env

### ISSUE THREAD

**Issue by mtaron** [Microsoft/azure-pipelines-tasks#9120]
The CondaEnvironment build task uses the "conda create" command. It would be *much* better from my perspective if it used "conda **env** create", or at least had the option to do so.

I create an [environment.yml](https://conda.io/docs/user-guide/tasks/manage-environments.html#create-env-file-manually) for each of my projects so that teammates can just run `conda env create -f environment.yml` to get all the dependencies they need for the specific project in an isolated environment. This works in a platform agnostic way. BUT `conda create` does not work with environment.yml files! `conda create --file` takes a "requirements.txt" style file that you can create by running `conda list --export > conda_requirements.txt`. These requirement files are platform specific. Obviously, I don’t want to have an environment.yml file and a conda_requirement.txt file that I regenerate every time I change the .yml file.

To work around this in Azure Pipelines, you could do something like this:

```yml
- script: $(CONDA)/Scripts/conda.exe env create --file environment.yml --name myenv
  displayName: Create conda environment

- task: CondaEnvironment@1
  displayName: Clone and activate conda environment
  inputs:
    updateConda: false
    createCustomEnvironment: true
    environmentName: ci
    packageSpecs: '' # empty string required; defaults to python=3, which conflicts with createOptions :(
    createOptions: '--clone myenv'
```

But due to #8504 , the CondaEnvironment task can't even be used for activation. :(

Once that issue has been addressed, it would much cleaner and faster (no extra clone) if the CondaEnvironment task supported env command directly.

**Comment by brcrista** [comment]
Thanks for the suggestion!  I agree that Anaconda's environment.yml is the best way to set up an Anaconda environment reproducibly, especially for CI.  I would expect that many projects using environment.yml would also be using azure-pipelines.yml.

We considered adding environment.yml support to the Conda Environment task.  This is what we ultimately decided: the Conda Environment task came from the pre-YAML days when you needed a task for everything.  YAML makes script easy, and just writing `script: conda env create -f environment.yml` is actually more concise than using the task!  Moreover, the task doesn't grow up well for other scenarios (such as installing from multiple channels), so we decided that we would start guiding users towards script for working with conda.  We updated our [Python guide](https://docs.microsoft.com/azure/devops/pipelines/languages/python#create-and-activate-an-anaconda-environment) to reflect that.

Do you agree with that?

**Comment by mtaron** [comment]
Hmm... I think that makes sense.

I see the "value add" of the CondaEnvironment task over a script in azure-pipelines.yml is that it (sorta) activates the environment for the rest of the job. But since it doesn't *really* activate the environment (not all environment variables set, no custom activation scripts run, etc.), it doesn't seem to add any value over a script...

Having some option to clone the environment from one script set and apply it to the next could solve a lot of problems here...

**Comment by brcrista** [comment]
That's been my thought as well.  Said another, it's become clear to me that getting the task to simulate all of the behavior of conda activate is too much, and we should focus on getting the actual `conda` to persist its environment changes in a pipeline.

**Comment by brcrista** [comment]
Thanks everyone for their feedback. We've decided to deprecate the Conda Environment task for the next release (see #9573).  Instead, we recommend invoking the `conda` CLI directly.

The task made sense when we were still focused on the old "Web designer"-based build definitions that lean heavily on tasks.  Today with YAML, script is much easier.

I created a guide with some YAML snippets here: https://aka.ms/pipelines-anaconda-guide.  Feedback welcome!
